src/client/i2p_service.h

```diff
@@ -33,68 +33,80 @@
 #ifndef SRC_CLIENT_I2P_SERVICE_H_
 #define SRC_CLIENT_I2P_SERVICE_H_
 
-#include <boost/asio.hpp>
 
 #include <atomic>
 #include <memory>
 #include <mutex>
 #include <string>
 #include <unordered_set>
 
-#include "destination.h"
-#include "identity.h"
+#include <boost/asio.hpp>
+
+#include "client/destination.h"
+#include "core/identity.h"
 
 namespace i2p {
 namespace client {
 
 class I2PServiceHandler;
+/// @class I2PService abstract class for service
+///// @brief base class for i2p service
 class I2PService {
  public:
+  /// @brief  constructor takes ClientDestination, defaults to null ptr
+  /// @param localDestination pointer to local destination
   explicit I2PService(
       std::shared_ptr<ClientDestination> localDestination = nullptr);
+  /// @brief  constructor takes a type of signing key
+  /// @param kt key type
   explicit I2PService(
       i2p::data::SigningKeyType kt);
+  /// @brief Destructor to clean up Handlers virtual
   virtual ~I2PService() { ClearHandlers(); }
-
+  /// @brief add a hander to set
+  /// @param conn I2pService pointer handler to add
   inline void AddHandler(
       std::shared_ptr<I2PServiceHandler> conn) {
     std::unique_lock<std::mutex> l(m_HandlersMutex);
     m_Handlers.insert(conn);
   }
-
+  /// @brief remove handler from set
+  /// @param conn I2PServiceHandler pointer to remove
   inline void RemoveHandler(
       std::shared_ptr<I2PServiceHandler> conn) {
     std::unique_lock<std::mutex> l(m_HandlersMutex);
     m_Handlers.erase(conn);
   }
-
+  /// @brief  clear out handlers from set
   inline void ClearHandlers() {
     std::unique_lock<std::mutex> l(m_HandlersMutex);
     m_Handlers.clear();
   }
-
+  /// @brief returns pointer to member m_LocalDestination
   inline std::shared_ptr<ClientDestination> GetLocalDestination() {
     return m_LocalDestination;
   }
-
+  /// @brief Set new member m_LocationDestination
+  /// @param dest pointer of type ClientDestination
   inline void SetLocalDestination(
       std::shared_ptr<ClientDestination> dest) {
     m_LocalDestination = dest;
   }
-
+  /// @brief Create a Stream to a destination
+  /// @param streamRequestComplete
   void CreateStream(
       StreamRequestComplete streamRequestComplete,
       const std::string& dest,
       int port = 0);
-
+  /// @brief return io_service refernce of member m_LocalDestination
```

Comment:
As a habit, you shouldn't put specific member names into comments because if/when they get refactored, you'll have out-dated comments (harder to maintain). Also, typo in *reference*.